storage/rocksdb/rdb_datadic.h

```diff
@@ -42,6 +42,48 @@ class Rdb_field_packing;
 class Rdb_cf_manager;
 class Rdb_ddl_manager;
 
+class Rdb_merge_operator : public rocksdb::AssociativeMergeOperator {
+ public:
+  virtual bool Merge(const rocksdb::Slice &key,
+                     const rocksdb::Slice *existing_value,
+                     const rocksdb::Slice &value, std::string *new_value,
+                     rocksdb::Logger *logger) const override {
+    
+    DBUG_ASSERT(new_value != nullptr);
+    std::string merged_value;
+
+    /* if no existing vaue, then persist the passed in value */
+    if (existing_value == nullptr) {
+      merged_value = Serialize(value);
+    } else {
+      merged_value = Deserialize(std::max(Serialize(value), Serialize(*existing_value)));
+    }
+
+    *new_value = merged_value;
+    return true;
+  }
+
+  virtual const char *Name() const override { return "Rdb_merge_operator"; }
+
+ private:
+  std::string Deserialize(const uint64_t data) const {
+    uchar value_buf[256] = {0};
```

Comment:
It might be simpler to allocate just 8 bytes for the uint64_t, instead of 256 bytes. Then you don't need to do the whole ptr += and ptr - value_buf calculations.